system-addon/content-src/components/TopSites/TopSite.jsx

```diff
@@ -110,7 +110,10 @@ class TopSite extends React.Component {
     }
   }
   onEditButtonClick() {
-    this.props.onEdit(this.props.index);
+    this.props.onEdit({
+      index: this.props.index,
+      link: this.props.link
```

Comment:
There's potential for wrong behavior if link is not the link at the index. I believe we should be able to look up the link via index from `TopSiteForm` with some `getState()`. Thoughts?

Reply:
👍